src/emit/binary.rs

```diff
@@ -1074,6 +1076,19 @@ impl<'a> Binary<'a> {
                 .const_int(error_with_loc.len() as u64, false),
         );
     }
+
+    /// Emit encoded error data of "Panic(uint256)" as interned global string.
+    pub(super) fn error_data_const(
+        &self,
+        ns: &Namespace,
+        code: PanicCode,
+    ) -> (PointerValue<'a>, IntValue<'a>) {
+        let bytes = create_encoder(ns, false).const_error_panic(code);
```

Comment:
Using codegen in emit is not a good idea. You could have an `Expression:StringLiteral` for this case. We usually assume that, when we reach emit, everything from CFG has already been constructed.

Reply:
I disagree. `create_encoder` doesn't take a `CFG` or anything, so it's impossible that it changes the CFG. By re-using it in emit, we use the _exact_ same code to create these constant values. I don't want to re-implement and test this again in emit.

Reply:
Perhaps the best here would be for `const_error_panic` to be a utility function that is not in the encoder trait. It transforms the panic code into bytes, while the encoder usually returns a Solidity array (an `Expression::Variable`).

This way, you can decouple the encoder from emit and still call that function in codegen.

Reply:
(My [above comment ](https://github.com/hyperledger/solang/pull/1449#discussion_r1271074040) relates to this as well)

Reply:
If a const encoder doesn't take a cfg then why can't it be used in emit?